packages/taquito-michelson-encoder/test/tokens/or.spec.ts

```diff
@@ -3,348 +3,1254 @@ import { createToken } from '../../src/tokens/createToken';
+          },
+        ],
+      },
+      0
+    ) as OrToken;
+    tokenOrWithOption = createToken(
+      {
+        prim: 'or',
+        args: [
+          { prim: 'int' },
+          {
+            prim: 'or',
+            args: [
+              { prim: 'nat' },
+              {
+                prim: 'or',
+                args: [
+                  { prim: 'mutez' },
+                  {
+                    prim: 'option',
+                    args: [
+                      {
+                        prim: 'or',
+                        args: [{ prim: 'int' }, { prim: 'string' }],
+                        annots: [],
+                      },
+                    ],
+                    annots: [],
+                  },
+                ],
+                annots: [],
+              },
+            ],
+            annots: [],
+          },
+        ],
+        annots: [],
+      },
+      0
+    ) as OrToken;
+  });
 
-    describe('EncodeObject', () => {
-        it('Should encode properly', () => {
-            expect(token.EncodeObject({ intTest: 10 })).toEqual({ prim: 'Left', args: [{ int: '10' }] });
-            expect(token.EncodeObject({ stringTest: '10' })).toEqual({ prim: 'Right', args: [{ string: '10' }] });
+  describe('EncodeObject', () => {
+    it('Should throw unless input is an object with single key', () => {
+      expect(() => tokenNoAnnots.EncodeObject(undefined)).toThrow(
+        'EncodeObject expectes an object with a single key but got:'
+      );
+      expect(() => tokenNoAnnots.EncodeObject(null)).toThrow(
+        'EncodeObject expectes an object with a single key but got:'
+      );
+      expect(() => tokenNoAnnots.EncodeObject([])).toThrow(
+        'EncodeObject expectes an object with a single key but got:'
+      );
+      expect(() => tokenNoAnnots.EncodeObject([10, 0])).toThrow(
+        'EncodeObject expectes an object with a single key but got:'
+      );
+      expect(() => tokenNoAnnots.EncodeObject({})).toThrow(
+        'EncodeObject expectes an object with a single key but got:'
+      );
+      expect(() => tokenNoAnnots.EncodeObject({ '0': 10, '1': '10' })).toThrow(
+        'EncodeObject expectes an object with a single key but got:'
+      );
+    });
 
-            expect(tokenNoAnnots.EncodeObject({ 0: 10 })).toEqual({ prim: 'Left', args: [{ int: '10' }] });
-            expect(tokenNoAnnots.EncodeObject({ 1: '10' })).toEqual({ prim: 'Right', args: [{ string: '10' }] });
+    it('For nested or with mixed annots, generateSchema, encodeObject and execute should be consistent', () => {
+      const schema2LevelMixedannots = token2LevelOrMixedAnnots.generateSchema();
+      expect(schema2LevelMixedannots).toEqual({
+        __michelsonType: 'or',
+        schema: {
+          '0': { __michelsonType: 'int', schema: 'int' },
+          '2': { __michelsonType: 'bool', schema: 'bool' },
+          A: { __michelsonType: 'nat', schema: 'nat' },
+        },
+      });
+      const leftLeft = {
```

Comment:
might be a good idea to put leftLeft, leftRight and right separately right above it's own expect assertion like token3LevelOrMixedAnnots examples line374